### DISCUSSION TITLE
Trouble deploying backend after project modification

### DISCUSSION THREAD

**Discussion by cjrieth** [microsoft/AI-Chat-App-Hack#75] [Q&A] [answered]
Hi! I have made some customizations to the AI chat app demo and it is now working to my liking when run locally with `app/start.sh`, so I have moved on to trying to deploy the backend to test its behavior. After running `azd up` I get the notification that the backend has successfully deployed,  yet when I try to access it I have no luck. Here is what the site looks like:

<img width="613" alt="image" src="https://github.com/microsoft/AI-Chat-App-Hack/assets/37090998/d7976626-7585-4905-9a46-3b3e65b46ad2">

Upon further inspection of the logs for the backend I see some error message that seem to indicate there are errors with module importing, I have attached the err logs: 

[2024_02_08_lw1sdlwk0003A7_default_docker[1].log](https://github.com/microsoft/AI-Chat-App-Hack/files/14203241/2024_02_08_lw1sdlwk0003A7_default_docker.1.log)

I am unsure about what to make of these errors or how to test out fixes, I would appreciate any help!

**Comment by pamelafox** [comment]
That's the classic error that we see when the server hasn't quite started. Do you still see errors after waiting 10ish minutes?

**Reply by cjrieth** [reply]
Went away for dinner and came back and it still does not appear to be working, same error posted.

I can access the server vis SSH and this is what I see:
<img width="1055" alt="image" src="https://github.com/microsoft/AI-Chat-App-Hack/assets/37090998/ff1321bd-f08f-42d2-baa5-6928026caa61">

`main.py` appears to be present, but it also looks like all of these files are within the `tmp` directory, not `home`. Could this be what is causing issues?

**Reply by pamelafox** [reply]
That's fine, that's what mine looks like as well. If you're using App Insights, can you double check that no exceptions got reported there? https://github.com/Azure-Samples/azure-search-openai-demo/?tab=readme-ov-file#monitoring-with-application-insights
Otherwise you might try clicking Restart server in the Portal.

**Reply by cjrieth** [reply]
Here is what App Insight shows me, looks like some failures surrounding GET requests for a text file?

<img width="1917" alt="image" src="https://github.com/microsoft/AI-Chat-App-Hack/assets/37090998/0c9370ec-b67a-4f8e-a6bb-f468af80fd06">

**Reply by pamelafox** [reply]
That's fine, to be expected. Can you try "Restart server" from Portal or even "azd deploy" from command line?

**Reply by cjrieth** [reply, marked as answer]
Seems like it was just some issue with the `us east` servers. I moved my resources to `us west` and it is all working fine.

Thank you @pamelafox and @john0isaac for the help!

**Comment by john0isaac** [comment]
Hey @cjrieth,

I tried deploying your main branch https://github.com/cjrieth/DubsBot and it worked.
![image](https://github.com/microsoft/AI-Chat-App-Hack/assets/64026625/35c5f724-2efe-44b2-82d3-2e3f08636ff1)


Note that initially, I did receive the same error you received but after waiting a couple of minutes everything worked successfully.

A solution I tried but I don't know if it affected the process or not is that I SSH'd inside Web Server and executed the command that was erroring.

```bash
python3 -m gunicorn main:app
```
The output tells me that the port is reserved which means that the app was already running.
But I recommend you give it a test and see. If you get anything other than whats in this screenshot please take a screenshot of it and post it here.
![image](https://github.com/microsoft/AI-Chat-App-Hack/assets/64026625/6f482008-44f7-49fa-8ac1-fd6aa13f12ce)


Also, note that getting this error is normal as `azd up` does two things `azd provision` creates cloud resources and `azd-deploy` uploads the app code to the cloud.
So during the provision stage your application may be already created but it's still empty so it tells you that the *main.py* is not found as it doesn't exist on the cloud yet. Once the app completely finishes provisioning it starts uploading your application code to the cloud.

My suggestion is:
- Do not stream the logs until the app is fully deployed.
- Ignore the old logs that say the main.py is not found.
- Check the latest logs after the app is deployed and test it like I showed you above.


Hope this helps!

**Reply by cjrieth** [reply]
What region did you deploy in?

**Reply by john0isaac** [reply]
Central US
